src/librustc_resolve/diagnostics.rs

```diff
@@ -197,6 +197,66 @@ See the Types section of the reference for more information about the primitive
 types:
 
 http://doc.rust-lang.org/reference.html#types
+"##,
+
+E0364: r##"
+Private items cannot be publicly re-exported.  This error indicates
+that you attempted to write a 'pub use' against a type or value that was
+not itself public.
+
+Here is an example that demonstrates the error:
+
+```
+mod foo {
+    const X: u32 = 1;
+}
+pub use foo::X;
+```
+
+The solution to this problem is to ensure that the items that you are
+re-exporting are themselves marked with 'pub':
+
+```
+mod foo {
+    pub const X: u32 = 1;
+}
+pub use foo::X;
+```
+
+See the Use Declarations section of the reference for more information
```

Comment:
The proper markdown would be to write

`[Use Declarations][1]`

and below

`[1]: http://doc.rust-lang.org/reference.html#use-declarations`

but I really dislike how it looks when reading plain output from the `--explain`. I think I was always right about being sceptical about use of markdown for these explanations.